src/server/v2.0/handler/scanexport.go

```
@@ -0,0 +1,237 @@
+
+	if err != nil {
+		return se.SendError(ctx, err)
+	}
+
+	jobID, err := se.scanDataExportCtl.Start(userContext, se.convertToCriteria(params.Criteria, secContext.GetUsername(), usr.UserID))
+	if err != nil {
+		return se.SendError(ctx, err)
+	}
+	scanDataExportJob.ID = jobID
+	return operation.NewExportScanDataOK().WithPayload(scanDataExportJob)
+}
+
+func (se *scanDataExportAPI) GetScanDataExportExecution(ctx context.Context, params operation.GetScanDataExportExecutionParams) middleware.Responder {
+	err := se.RequireAuthenticated(ctx)
+	if err != nil {
+		return se.SendError(ctx, err)
+	}
+	execution, err := se.scanDataExportCtl.GetExecution(ctx, params.ExecutionID)
+	if err != nil {
+		return se.SendError(ctx, err)
+	}
+	sdeExec := models.ScanDataExportExecution{
+		EndTime:     strfmt.DateTime(execution.EndTime),
+		ID:          execution.ID,
+		StartTime:   strfmt.DateTime(execution.StartTime),
+		Status:      execution.Status,
+		StatusText:  execution.StatusMessage,
+		Trigger:     execution.Trigger,
+		UserID:      execution.UserID,
+		JobName:     execution.JobName,
+		UserName:    execution.UserName,
+		FilePresent: execution.FilePresent,
+	}
+
+	return operation.NewGetScanDataExportExecutionOK().WithPayload(&sdeExec)
+}
+
+func (se *scanDataExportAPI) DownloadScanData(ctx context.Context, params operation.DownloadScanDataParams) middleware.Responder {
+	err := se.RequireAuthenticated(ctx)
+	if err != nil {
+		return se.SendError(ctx, err)
+	}
+	execution, err := se.scanDataExportCtl.GetExecution(ctx, params.ExecutionID)
+	if err != nil {
+		se.SendError(ctx, err)
+	}
+
+	// check if the execution being downloaded is owned by the current user
+	secContext, err := se.GetSecurityContext(ctx)
+	if err != nil {
+		return se.SendError(ctx, err)
+	}
+
+	if secContext.GetUsername() != execution.UserName {
+		return middleware.ResponderFunc(func(writer http.ResponseWriter, producer runtime.Producer) {
```

here we can just return an forbidden error?

---

Ack. Will pick this in a subsequent CL

---

Done